test/com/facebook/buck/jvm/java/JavaAnnotationProcessorDescriptionTest.java

```diff
@@ -0,0 +1,200 @@
+  }
+
+  @Test
+  public void testOnlyProcessorClassIsPassedToJavaAnnotationProcessor() {
+    // Given
+    JavaAnnotationProcessorDescriptionArg arg =
+        JavaAnnotationProcessorDescriptionArg.builder()
+            .setName("annotation_processor")
+            .setIsolateClassLoader(false)
+            .setDoesNotAffectAbi(true)
+            .setSupportsAbiGenerationFromSource(true)
+            .setProcessorClass(Optional.of("Needle.HayStack"))
+            .setProcessorClasses(ImmutableSet.of("Foo.Bar", "Bar.Foo"))
+            .build();
+
+    BuildRuleResolver buildRuleResolver = new TestBuildRuleResolver();
+    ProjectFilesystem projectFilesystem = new FakeProjectFilesystem();
+
+    BuildTarget buildTarget = BuildTargetFactory.newInstance("//foo:baz");
+
+    BuildRuleParams params =
+        TestBuildRuleParams.create().withDeclaredDeps(buildRuleResolver.getAllRules(arg.getDeps()));
+
+    // When
+    JavaAnnotationProcessor javaAnnotationProcessor =
+        (JavaAnnotationProcessor)
+            new JavaAnnotationProcessorDescription()
+                .createBuildRule(
+                    TestBuildRuleCreationContextFactory.create(
+                        buildRuleResolver, projectFilesystem),
+                    buildTarget,
+                    params,
+                    arg);
+
+    // Verify
+    JavacPluginProperties props =
+        JavacPluginProperties.builder()
+            .setCanReuseClassLoader(true)
+            .setDoesNotAffectAbi(true)
+            .setSupportsAbiGenerationFromSource(true)
+            .addProcessorNames("Needle.HayStack")
+            .build();
+
+    assertEquals(javaAnnotationProcessor.getUnresolvedProperties(), props);
+  }
+
+  @Test(expected = HumanReadableException.class)
+  public void testRaisesExceptionWhenNoProcessorClassIsSpecified() {
+    // Given
+    JavaAnnotationProcessorDescriptionArg arg =
+        JavaAnnotationProcessorDescriptionArg.builder()
+            .setName("annotation_processor")
+            .setIsolateClassLoader(false)
+            .setDoesNotAffectAbi(true)
+            .setSupportsAbiGenerationFromSource(true)
+            .build();
+
+    BuildRuleResolver buildRuleResolver = new TestBuildRuleResolver();
+    ProjectFilesystem projectFilesystem = new FakeProjectFilesystem();
+
+    BuildTarget buildTarget = BuildTargetFactory.newInstance("//foo:baz");
+
+    BuildRuleParams params =
+        TestBuildRuleParams.create().withDeclaredDeps(buildRuleResolver.getAllRules(arg.getDeps()));
+
+    // When
+    JavaAnnotationProcessor javaAnnotationProcessor =
+        (JavaAnnotationProcessor)
+            new JavaAnnotationProcessorDescription()
+                .createBuildRule(
+                    TestBuildRuleCreationContextFactory.create(
+                        buildRuleResolver, projectFilesystem),
+                    buildTarget,
+                    params,
+                    arg);
+
+    // Verify
+    // HumanReadableException is thrown.
```

Comment:
Can you verify the actual exception message? Otherwise this test may continue to work when there is a different problem.

Reply:
sure, done.